docker-compose.yml

```diff
@@ -806,6 +808,7 @@ services:
     command: foreman run worker
     depends_on:
       - diet-error-handler
+      - redis
```

Comment:
Do we need this because I thought `<< : *whitehall` would include this dependency? If not, I reckon we'll also need `mysql` (for Whitehall anyway) because the workers will interact with the database.

Reply:
Hmmmm I suspect that our `depends_on` property is [overriding](https://stackoverflow.com/questions/53965417/merging-yaml-arrarys-nested) the one defined in `whitehall`, rather than merging with it. That would explain why we're seeing these errors and why this PR fixes them.

Not sure what the wider implications of that are. The only other arrays defined there are `links` and `ports`, which aren't overwritten (with the exception of `whitehall-worker`, which [overwrites `ports` to an empty array](https://github.com/alphagov/publishing-e2e-tests/blob/a948fe46388c482ebd10e6807173dbf78d4beafd/docker-compose.yml#L818). This _also_ suggests that properties are overridden rather than merged).

Reply:
I wondered if that's the case. I suspect (for Whitehall anyway) we'll also need to add a link to `mysql` here, but this is a good start.

Reply:
Yeah, I'm pretty sure yaml doesn't merge values when you do a `<<`